modules/promotions/admin-menu-items/custom-fonts-promotion-item.php

```diff
@@ -15,18 +17,47 @@ public function get_page_title() {
 		return esc_html__( 'Custom Fonts', 'elementor' );
 	}
 
+	public function get_image_url() {
```

Comment:
Hey @Barcov 

While there is nothing inherently wrong with using the same filter multiple times, I do believe we can refactor the `base-promotion-item.php` a bit and make this more efficient.

Can you try and change the `render()` method to something like this:

```
public function render() {
		$data = [
			'image_url' => $this->get_image_url(),
			'promotion_title' => $this->get_promotion_title(),
			'cta_url' => $this->get_cta_url(),
			'cta_text' => $this->get_cta_text(),
		];

		$data = apply_filters( 'elementor/promotion/' . $this->get_name() . '/custom_promotion', $data );
		?>
		<div class="wrap">
			<div class="elementor-blank_state">
				<img src="<?php echo esc_url( $data['image_url'] ); ?>" loading="lazy" />

				<h3><?php Utils::print_unescaped_internal_string( $data['promotion_title'] ); ?></h3>

				<p><?php $this->render_promotion_description(); ?></p>

				<a class="elementor-button go-pro" href="<?php echo esc_url( $data['cta_url'] ); ?>">
					<?php Utils::print_unescaped_internal_string( $data['cta_text']  ); ?>
				</a>
			</div>
		</div>
		<?php
}
```
I am basically just building the `$data` array with the methods from the child classes, and then running the filter once to override all the array. And then instead of calling those methods inside the html I just use `$data['whatever']`.

`$this->render_promotion_description()` will have to change/remove - I am not sure why it was done like this in the first place. (To just echo something).
`get_name()` has to be introduced in the base class and implemented in the child classes where it's needed.

WDYT?